config_utilities.hpp

```diff
@@ -0,0 +1,1031 @@
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice, this
+   list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its contributors
+   may be used to endorse or promote products derived from this software without
+   specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * Depending on which headers are available, ROS dependencies are included in
+ * the library. Make sure to include config_utilities.hpp after these headers.
+ */
+
+// <ros/node_handle.h>
+#ifdef ROSCPP_NODE_HANDLE_H
+#ifndef CONFIG_UTILITIES_ROS_ENABLED
+#define CONFIG_UTILITIES_ROS_ENABLED
+#endif
+#endif
+
+// <kindr/minimal/quat-transformation.h>
+#ifdef KINDR_MINIMAL_QUAT_TRANSFORMATION_H_
+#ifndef CONFIG_UTILITIES_TRANSFORMATION_ENABLED
+#define CONFIG_UTILITIES_TRANSFORMATION_ENABLED
+#endif
+#endif
+
+
+#ifndef CONFIG_UTILITIES_CORE_HPP_
+#define CONFIG_UTILITIES_CORE_HPP_
+
+#include <iomanip>
+#include <map>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <type_traits>
+#include <typeinfo>
+#include <unordered_map>
+#include <utility>
+
+#include <glog/logging.h>
+#include <xmlrpcpp/XmlRpcValue.h>
+
+namespace config_utilities {
+
+/**
+ * ==================== Settings ====================
+ */
+namespace internal {
+struct Settings {
+  Settings(const Settings &other) = delete;
+  Settings &operator=(const Settings &other) = delete;
+
+  // Settings
+  unsigned int default_print_width = 80;
+  unsigned int default_print_indent = 30;
+  unsigned int default_subconfig_indent = 3;
+
+  static Settings &instance() {
+    static Settings settings;
+    return settings;
+  }
+
+private:
+  Settings() = default;
+};
+} // namespace internal
+
+// Access.
+internal::Settings &Settings() { return internal::Settings::instance(); }
```

Comment:
Are you sure you need this? It feels like you would achieve the same as the above by simply making all members of `Settings` static. You could then access them with `GlobalSettings::default_print_width = 80`, which might be a bit clearer to read and it's a bit simpler since you don't need a (fancy) accessor to a static instance of `Settings` in `Settings`.